fuzz: reported crash in mempool during nightly run

Possibly reordered some initialization code during #7171?

Actions run: https://github.com/tendermint/tendermint/actions/runs/1410627573

Log:

```
panic: testing: Verbose called before Init

goroutine 1 [running]:
testing.Verbose(...)
	/opt/hostedtoolcache/go/1.17.2/x64/src/testing/testing.go:453
github.com/tendermint/tendermint/libs/log.TestingLoggerWithOutput({0xc0000001a0, 0xc000048000})
	/home/runner/work/tendermint/tendermint/libs/log/testing.go:34 +0x465
github.com/tendermint/tendermint/libs/log.TestingLogger(...)
	/home/runner/work/tendermint/tendermint/libs/log/testing.go:23
github.com/tendermint/tendermint/test/fuzz/mempool.init.0.func1()
	/home/runner/work/tendermint/tendermint/test/fuzz/mempool/checktx.go:31 +0x85
github.com/tendermint/tendermint/test/fuzz/mempool.Fuzz({0x7fa45be00000, 0x0, 0x0})
	/home/runner/work/tendermint/tendermint/test/fuzz/mempool/checktx.go:43 +0x4c
go-fuzz-dep.Main({0xc0001f7f68, 0x1, 0xcc1880})
	go-fuzz-dep/main.go:36 +0x15b
main.main()
	github.com/tendermint/tendermint/test/fuzz/mempool/go.fuzz.main/main.go:15 +0x3b
exit status 2
```

--- a/test/fuzz/mempool/checktx.go
+++ b/test/fuzz/mempool/checktx.go
@@ -2,6 +2,7 @@ package mempool
 
 import (
 	"context"
+	"testing"
 
 	abciclient "github.com/tendermint/tendermint/abci/client"
 	"github.com/tendermint/tendermint/abci/example/kvstore"
@@ -25,6 +26,7 @@ func init() {
 	cfg := config.DefaultMempoolConfig()
 	cfg.Broadcast = false
 
+	testing.Init()
 	getMp = func() mempool.Mempool {
 		if mp == nil {
 			mp = mempool.NewTxMempool(

--- a/test/fuzz/mempool/checktx.go
+++ b/test/fuzz/mempool/checktx.go
@@ -2,7 +2,6 @@ package mempool
 
 import (
 	"context"
-	"testing"
 
 	abciclient "github.com/tendermint/tendermint/abci/client"
 	"github.com/tendermint/tendermint/abci/example/kvstore"
@@ -26,11 +25,10 @@ func init() {
 	cfg := config.DefaultMempoolConfig()
 	cfg.Broadcast = false
 
-	testing.Init()
 	getMp = func() mempool.Mempool {
 		if mp == nil {
 			mp = mempool.NewTxMempool(
-				log.TestingLogger().With("module", "mempool"),
+				log.NewNopLogger(),
 				cfg,
 				appConnMem,
 				0,

